### PR TITLE
Adding the ability to import trait values from a list

### PR DESCRIPTION
**Pull request by jnwng**
addresses https://github.com/theskeletoncrew/treat-toolbox/issues/14

lets the user import a list of comma-separated trait value / rarity pairs instead of having to do them each manually.

valid files look like:
```csv
red,0.05
blue,0.05
green,0.9
```
we'll also do a check to make sure `rarity` adds up to 1. 

https://user-images.githubusercontent.com/1118534/143933548-d3c84b37-3f31-4bec-9b4c-cdb3cb65a20a.mov

open questions:
- [ ] as it stands, this will overwrite trait values that exist already. should we warn when we're doing that instead of blowing them away?
- [ ] do we need to do any more validation on the trait values? i have found that having a combination of machine-readable and human-readable trait values are nice, but that's not currently captured in the Trait model

### CODE REVIEW

**Comment by jverdi** [comment]
@jnwng thanks! this is great.

Some minor feedback:
- if there is an extra newline at the end of the file, that line's rarity gets interpreted as NaN, which causes the whole file to get rejected
- Lets move the sample csv files to be in their own directory alongside `test-images` as something like `test-trait-csvs`

To answer open questions:
- I think its fine to overwrite the existing traits by default, but asking for confirmation in a dialog feels like it would be a good idea

Could you add a bit more detail on your second question? I'm not sure I followed

**Comment by cosimo-rip** [comment]
@jnwng thanks! this is great.

Some minor feedback:

if there is an extra newline at the end of the file, that line's rarity gets interpreted as NaN, which causes the whole file to get rejected
Lets move the sample csv files to be in their own directory alongside test-images as something like test-trait-csvs
To answer open questions:

I think its fine to overwrite the existing traits by default, but asking for confirmation in a dialog feels like it would be a good idea
Could you add a bit more detail on your second question? I'm not sure I followed

File: pages/projects/[projectId]/collections/[collectionId]/traits/[traitId]/values/import-list.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                            accept=".csv"
+                            onChange={(e) => setSelectedFiles(e.target.files)}
+                          />
+                          <p className="text-xs text-gray-500 pb-5">
+                            Please select a file with comma-separated values.
+                          </p>
+                          <p className="text-xs text-gray-500 pb-5">
+                            {/* {selectedFiles
+                              ? "[" + selectedFiles?.length + " files selected]"
+                              : ""} */}
```
Comment (cosimo-rip) [inline pages/projects/[projectId]/collections/[collectionId]/traits/[traitId]/values/import-list.tsx:202]: i think it makes sense to uncomment this section too - it helps to show that something happened after selecting a file

Comment (jnwng) [reply pages/projects/[projectId]/collections/[collectionId]/traits/[traitId]/values/import-list.tsx]: done!

**Comment by jnwng** [comment]
> if there is an extra newline at the end of the file, that line's rarity gets interpreted as NaN, which causes the whole file to get rejected

resolved, i'm now filtering out empty lines in `papaparse`.

> Lets move the sample csv files to be in their own directory alongside test-images as something like test-trait-csvs

✅

**open questions**
> I think its fine to overwrite the existing traits by default, but asking for confirmation in a dialog feels like it would be a good idea

done, also realized the import action was not upserting when the value already existed. in this case when there's a collision we'll take the value from the uploaded file as the source of truth; what should happen with existing trait values that are not in the CSV? should we just blow away everything and assume the CSV file is the total source of truth since all the rarities need to add up to 1 anyway?

re: the second question, the perspective here was whether or not we wanted to add another field to the trait that is "human readable" (e.g., i typically have kebab-cased traits for code purposes but when i display them i want something else more interesting. out of the scope of this PR though!

**Comment by cosimo-rip** [comment]
> what should happen with existing trait values that are not in the CSV? should we just blow away everything and assume the CSV file is the total source of truth since all the rarities need to add up to 1 anyway?

yeah - this was the only issue i came across in testing. lets have it blow them all away in that case, and maybe just change the wording in the confirmation dialog to indicate that is what is about to happen.

so instead of:
"Are you sure you want to overwrite the existing trait values and rarities? This action cannot be undone."

maybe:
"Are you sure you want to replace all existing trait values and rarities with those in the uploaded list? This action cannot be undone."

**Comment by jnwng** [comment]
> > what should happen with existing trait values that are not in the CSV? should we just blow away everything and assume the CSV file is the total source of truth since all the rarities need to add up to 1 anyway?
> 
> @jnwng yeah - this was the only issue i came across in testing. lets have it blow them all away in that case, and maybe just change the wording in the confirmation dialog to indicate that is what is about to happen.
> 
> so instead of: "Are you sure you want to overwrite the existing trait values and rarities? This action cannot be undone."
> 
> maybe: "Are you sure you want to replace all existing trait values and rarities with those in the uploaded list? This action cannot be undone."

done!

**Comment by skeletoncrewrip** [comment]
thanks!

**Pull request merged**